### PR TITLE
[VL] Fix fallback caused by filter of array/map column type

### PR DESCRIPTION
**Pull request by zuochunwei**
## What changes were proposed in this pull request?

(Please fill in changes proposed in this fix)

(Fixes: \#ISSUE-ID)

## How was this patch tested?

(Please explain how this patch was tested. E.g. unit tests, integration tests, manual tests)

(If this patch involves UI changes, please attach a screenshot; otherwise, remove this)

### CODE REVIEW

File: cpp/velox/substrait/SubstraitToVeloxPlan.cc [hunk 1]
```diff
@@ -1743,19 +1749,15 @@ void SubstraitToVeloxPlanConverter::constructSubfieldFilters(
...
     if (rangeSize == 0 && !nullAllowed) {
-      // TODO
-      if (fileFormat_ == facebook::velox::dwio::common::FileFormat::ORC) {
-        // uncomment the following line when Velox decimal support IsNotNull filter
-        // filters[common::Subfield(inputName, true)] = std::move(std::make_unique<common::IsNotNull>());
-        VELOX_NYI("constructSubfieldFilters not support IsNotNull for input type '{}'", inputType);
-      } else {
-        VELOX_NYI("constructSubfieldFilters not support IsNotNull for input type '{}'", inputType);
-      }
+      filters[common::Subfield(inputName, true)] = std::move(std::make_unique<common::IsNotNull>());
```
Comment (rui-mo) [inline cpp/velox/substrait/SubstraitToVeloxPlan.cc:1757]: > bool nullAllowed = filterInfo.nullAllowed_;

Do we need to obey the nullAllowed info?

File: cpp/velox/substrait/SubstraitToVeloxPlan.cc [hunk 2]
```diff
@@ -1560,6 +1560,12 @@ void SubstraitToVeloxPlanConverter::setFilterInfo(
         }
       }
       break;
+    case TypeKind::ARRAY:
+      // Doing nothing here can let filter sIsNotNull still work.
```
Comment (rui-mo) [inline cpp/velox/substrait/SubstraitToVeloxPlan.cc:1564]: sIsNotNull -> IsNotNull?

File: cpp/velox/substrait/SubstraitToVeloxPlan.cc [hunk 3]
```diff
@@ -1560,6 +1560,12 @@ void SubstraitToVeloxPlanConverter::setFilterInfo(
...
+      break;
+    case TypeKind::MAP:
+      // Doing nothing here can let filter sIsNotNull still work.
+      break;
```
Comment (rui-mo) [inline cpp/velox/substrait/SubstraitToVeloxPlan.cc:1568]: Same.

File: cpp/velox/substrait/SubstraitToVeloxPlan.cc (hunk 1 above)
Comment (rui-mo) [reply cpp/velox/substrait/SubstraitToVeloxPlan.cc:1757]: Maybe a better way to achieve this is to separate the unsupported filters to remaining filter in `separateFilters`. Only filters can be pushed down are sent to `constructSubfieldFilters`.

File: cpp/velox/substrait/SubstraitToVeloxPlan.cc (hunk 2 above)
Comment (zuochunwei) [reply cpp/velox/substrait/SubstraitToVeloxPlan.cc:1564]: OK

File: cpp/velox/substrait/SubstraitToVeloxPlan.cc [hunk 4]
```diff
@@ -1743,19 +1749,15 @@ void SubstraitToVeloxPlanConverter::constructSubfieldFilters(
   uint32_t rangeSize = std::max(filterInfo.lowerBounds_.size(), filterInfo.upperBounds_.size());
 
   if constexpr (KIND == facebook::velox::TypeKind::HUGEINT) {
+    // TODO: open here when the Velox's modification is ready
```
Comment (rui-mo) [inline cpp/velox/substrait/SubstraitToVeloxPlan.cc:1752]: Nit: add `.` at the end for comments.

File: cpp/velox/substrait/SubstraitToVeloxPlan.cc [hunk 5]
```diff
@@ -1743,19 +1749,15 @@ void SubstraitToVeloxPlanConverter::constructSubfieldFilters(
...
+    VELOX_NYI("constructSubfieldFilters not support for HUGEINT type");
+  } else if constexpr (KIND == facebook::velox::TypeKind::ARRAY || KIND == facebook::velox::TypeKind::MAP) {
+    // Only IsNotNull filter is supported for the above two type kinds now
```
Comment (rui-mo) [inline cpp/velox/substrait/SubstraitToVeloxPlan.cc:1755]: Same.

File: cpp/velox/substrait/SubstraitToVeloxPlan.cc (hunk 1 above)
Comment (zuochunwei) [reply cpp/velox/substrait/SubstraitToVeloxPlan.cc:1757]: sounds better, but how about keeping this change for this PR, and proposing a new PR to refactor this.

File: cpp/velox/substrait/SubstraitToVeloxPlan.cc (hunk 4 above)
Comment (zuochunwei) [reply cpp/velox/substrait/SubstraitToVeloxPlan.cc:1752]: OK

File: cpp/velox/substrait/SubstraitToVeloxPlan.cc (hunk 5 above)
Comment (zuochunwei) [reply cpp/velox/substrait/SubstraitToVeloxPlan.cc:1755]: OK

**Review by rui-mo — APPROVED** [review]

**Pull request merged**